src/main/java/seedu/contax/logic/commands/BatchCommand.java

```diff
@@ -48,29 +56,43 @@
      * @param searchType                search type of field apply matcher
      * @param userInput                 regex provided by user
      */
-    public BatchCommand(String commandInput, SearchType searchType, String userInput) {
+    public BatchCommand(String commandInput, SearchType searchType, ArgumentMultimap userInput) {
         requireNonNull(commandInput);
         requireNonNull(searchType);
         requireNonNull(userInput);
         this.commandInput = commandInput;
         this.searchType = searchType;
-        this.userInput = userInput.trim();
+        this.userInput = userInput;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Index> indexList = matchInputStringToIndex(model, searchType, userInput);
+        String userValue = "";
+        List<Index> indexList = new ArrayList<>();
+        if (userInput.getValue(PREFIX_EQUALS).isPresent()) {
+            userValue = userInput.getValue(PREFIX_EQUALS).get();
+            indexList = matchInputStringToIndex(model, searchType, userValue, "=");
+        } else if (userInput.getValue(PREFIX_START_WITH).isPresent()) {
+            userValue = userInput.getValue(PREFIX_START_WITH).get();
+            indexList = matchInputStringToIndex(model, searchType, userValue, "^");
+        } else if (userInput.getValue(PREFIX_END_WITH).isPresent()) {
+            userValue = userInput.getValue(PREFIX_END_WITH).get();
+            indexList = matchInputStringToIndex(model, searchType, userValue, "$");
+        } else {
+            throw new CommandException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, BatchCommand.MESSAGE_USAGE));
```

Comment:
You should not be parsing in the command class.

Reply:
thx I will move to parsing instead